zulip/integrations/confluence/confluence.py

```diff
@@ -0,0 +1,145 @@
+
+[Confluence_api]
+username =
+consumer_key =
+domain_Url =
+
+
+In order to obtain a consumer key & secret, you must register a
+new application under your Confluence account:
+
+1 Log in to https://id.atlassian.com.
+2 Click API tokens, then Create API token.
+3 Use Copy to clipboard, and paste the token to your script
+  or elsewhere:
+
+"""
+parser = zulip.add_default_arguments(argparse.ArgumentParser("Fetch space and page from Confluence."))
+parser.add_argument('--instructions',
+                    action='store_true',
+                    help='Show instructions for the confluence bot setup and exit'
+                    )
+opts = parser.parse_args()
+if opts.instructions:
+    print(INSTRUCTIONS)
+    sys.exit()
+if opts.zulip_config_file:
+    zuliprc_path = opts.zulip_config_file
+
+try:
+    config = ConfigParser()
+    config.read(CONFIGFILE)
+    username = config.get('Confluence_api', 'username')
+    consumer_key = config.get('Confluence_api', 'consumer_key')
+    domain_Url = config.get('Confluence_api', 'domain_Url')
+except Exception:
+    parser.error("Please provide a ~/.zulip_confluencerc")
+
+local_file = "zulip.txt"
+f = open(local_file, "w+")
+f.write('0')
+f.close()
+
+def confluence_api() -> None:
+
+    params = ('type', 'page')
+    base_url = 'https://{}/wiki/rest/api/content'.format(domain_Url)
+    auth = (username, consumer_key)
+    response = requests.get(base_url, params=params, auth=auth)
+    res = response.json()
+    # length of json data
+    length_api = len(res['results'])
+    # Fetch value from database models
+    f = open(local_file, "r")
+    length_confluence_data = int(f.read())
+    # compare the lenth of json from database to the new
+    # length of json which is fetch from api
+    # if length of fetch json from api  is greater than
+    # save length then we call zulip api
+    if (length_api > length_confluence_data):
+        test1_json = (res['results'][length_api-1]['_links']['webui'])
+        title = res['results'][length_api-1]['title']
+        get_spaces_and_pages = spacepage(test1_json)
+        format_page_and_string = "New {} is created with title {}".format(get_spaces_and_pages, title)
+        client = zulip.Client(config_file = zuliprc_path)
+        request = {
+            "type": "stream",
+            "to": "confluence",
+            "subject": "Page",
+            "content": format_page_and_string
+        }
+        result = client.send_message(request)
+        f = open(local_file, "w")
+        f.write(str(length_api))
+    # if user delete pages or space it become update in database
+    elif (length_api < length_confluence_data):
+        f = open(local_file, "w")
+        f.write(str(length_api))
+    return (None)
+# here we check if there is updation of space or page
+def spacepage(get_spaces_and_pages: str) -> str:
+    page_space = get_spaces_and_pages.split('/')
+    if ('pages' in page_space):
+        return ('page\n')
```

Comment:
`if` and `return` statements followed by conditionals enclosed in parenthesis ( `()` ) are not necessary in Python. Again, I would recommend reading PEP8 for more information about style and conventions in Python.